printf/stdio_printf.c

```diff
@@ -0,0 +1,2645 @@
+/*
+ * Phoenix-RTOS
+ *
+ * POSIX.1-2017 standard library functions tests
+ * HEADER:
+ *    - stdio.h
+ *
+ * TESTED:
+ *    - dprintf()
+ *    - fprintf()
+ *    - printf()
+ *    - snprintf()
+ *    - sprintf()
+ *    - vdprintf()
+ *    - vfprintf()
+ *    - vprintf()
+ *    - vsnprintf()
+ *    - vsprintf()
+ *
+ * Copyright 2023 Phoenix Systems
+ * Author: Mateusz Bloch
+ *
+ * This file is part of Phoenix-RTOS.
+ *
+ * %LICENSE%
+ */
+
+#include <stdio.h>
+#include <stdarg.h>
+#include <stdlib.h>
+#include <stdint.h>
+#include <string.h>
+#include <float.h>
+#include <limits.h>
+#include <stdbool.h>
+#include <unistd.h>
+#include <fcntl.h>
+#include <errno.h>
+#include <unity_fixture.h>
+#include <signal.h>
+
+static FILE *output_file;
+static char buff[700];
+
+enum OutputType { DTEST,
+	FTEST,
+	TEST,
+	SNTEST,
+	STEST };
+
+#define NUM_OUTPUT_TYPES 5
+
+#define PATH "stdio_printf_test"
+
+/* comma delimiter macro ',' for multiple arguments parsing to *printf() function via macros */
+#define COMMA ,
+
+/* macro assertions*/
+
+#define printf_test(format, value, expect) \
+	{ \
+		for (enum OutputType type = DTEST; type < NUM_OUTPUT_TYPES; type++) { \
+			memset(buff, 0, sizeof(buff)); \
+			rewind(output_file); \
+			switch (type) { \
+				case DTEST: \
+					dprintf(fileno(output_file), format, value); \
+					break; \
+				case FTEST: \
+					fprintf(output_file, format, value); \
+					break; \
+				case TEST: { \
+					int stdout_fd = dup(STDOUT_FILENO); \
+					int fd = open(PATH, O_WRONLY | O_CREAT, 0666); \
+					dup2(fd, STDOUT_FILENO); \
+					printf(format, value); \
+					fflush(stdout); \
+					dup2(stdout_fd, STDOUT_FILENO); \
+					close(fd); \
+					close(stdout_fd); \
+					break; \
+				} \
+				case SNTEST: \
+					snprintf(buff, strlen(expect) + 1, format, value); \
+					break; \
+				case STEST: \
+					sprintf(buff, format, value); \
+					break; \
```

Comment:
We use braces even for one line blocks.

```suggestion
				case SNTEST: \
					snprintf(buff, strlen(expect) + 1, format, value); \
					break; \
				} \
				case STEST: { \
					sprintf(buff, format, value); \
					break; \
				} \
```

Reply:
please note that adding subscopes (`{ }`) in switch `case's` is not needed (unless You're declaring variables there) and IMHO decreases readability.

Reply:
Yeah right, it isn't needed there, we will remove them.